include/llvm/DebugInfo/DWARF/DWARFDebugAranges.h

```diff
@@ -32,20 +32,21 @@ class DWARFDebugAranges {
   void construct();
 
   struct Range {
-    explicit Range(uint64_t LowPC = -1ULL, uint64_t HighPC = -1ULL,
-                   uint32_t CUOffset = -1U)
-      : LowPC(LowPC), Length(HighPC - LowPC), CUOffset(CUOffset) {}
+    explicit Range(uint64_t LowPC = std::numeric_limits<uint64_t>::max(),
+                   uint64_t HighPC = std::numeric_limits<uint64_t>::max(),
+                   uint32_t CUOffset = -std::numeric_limits<uint32_t>::max())
```

Comment:
In some places you transform -1U into a uint32_t type and in others to just unsigned.
Is this just based on the lhs variable type? 
Could one make the argument that an implicit conversion should take place from, say, an unsigned here into a uint32_t?